sunpy/net/dataretriever/client.py

```diff
@@ -225,7 +170,57 @@ def _can_handle_query(cls, *query):
         `sunpy.net.fido_factory.UnifiedDownloaderFactory`
         class uses to dispatch queries to this Client.
         """
-        raise NotImplementedError
+        adict = cls.register_values()
+        if cls.optional == set():
+            cls.optional = {k for k in adict.keys()} - cls.required
+        if not cls.check_attr_types_in_query(query, cls.required, cls.optional):
+            return False
+        for key in adict:
+            all_vals = [i[0].lower() for i in adict[key]]
+            for x in query:
+                if isinstance(x, key) and str(x.value).lower() not in all_vals:
+                    return False
+        return True
+
+    def post_search_hook(self, exdict, matchdict):
+        """
+        Helper function used after `:func:~sunpy.net.dataretriever.GenericClient.search`
+        which makes the extracted metadata representable in a query response table.
+
+        Parameters
+        ----------
+        exdict: `dict`
+            Represents metadata extracted from files.
+        matchdict: `dict`
+            Contains attr values accessed from `register_values()`
+            and the search query itself.
+
+        Returns
+        -------
+        map_: `~collections.OrderedDict`
+            An Ordered Dictionary which is used by `QueryResponse`
+            to show results.
+        """
+        map_ = OrderedDict()
+        almost_day = TimeDelta(1 * u.day - 1 * u.millisecond)
+        if 'month' in exdict and 'day' in exdict:
+            start = parse_time("{}/{}/{}".format(exdict['year'], exdict['month'], exdict['day']))
+            end = start + almost_day
+        elif 'year' in exdict:
+            start = parse_time("{}/{}/{}".format(exdict['year'], 1, 1))
+            end = parse_time("{}/{}/{}".format(exdict['year'], 12, 31)) + almost_day
+        map_['Time'] = TimeRange(start, end)
+        map_['Start Time'] = start.strftime(TIME_FORMAT)
+        map_['End Time'] = end.strftime(TIME_FORMAT)
+        map_['Instrument'] = matchdict['Instrument'][0].upper()
+        if 'Physobs' in matchdict:
```

Comment:
why only the if statement for physobs, what if my new client doesn't define a source, provider or even instrument? This whole method feels like it's pretty specialised for what exists at the moment rather than the design of the `GenericClient`.

Am I right in thinking that this is a converter from `cls.pattern` to a response table? Can we not generalise this by restricting (via docs) the possible names of parameters in `cls.pattern`?

Reply:
In fact, couldn't we use the fields present in the pattern to know which things we need to process here?

Reply:
Some fields are not present in the `pattern`, like `Source`, `Instrument`, `Provider` etc. So we can't do that way. I'm using matchdict for that work. It uses `register_values` for getting those attrs. The problem is some things may not be even registered as attrs. (Like Time and Wavelength). So we need an individual compare after running loop over the things available from `cls.pattern`.

Yes you are right I shoudn't check only for `Physobs`. Infact I should check for all other attrs too.

Reply:
This brings up an interesting thought about how we validate `register_values` and the pattern. If things aren't in the pattern they should have one value in `register_values` and then all the results take that value? If they have more than one value in `register_values` then they should be in the pattern?

Can you not use this logic to determine how to build the results? It would then be much more flexible. I wouldn't expect you to need to call out anything explicitly in this method other than maybe time?

Reply:
Yes, not assuming existence of specific attrs now. Applied that logic in `post_search_hook`